### PR TITLE
Certificates onboarding screen (EXPOSUREAPP-7435)

### PR DESCRIPTION
**Pull request by AlexanderAlferov**
Certificates onboarding screen.

Screenshot 2021-05-27 at 18 52 35

Whats left:
- [ ] Accessibility illustration description

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/main/MainActivity.kt [hunk 1]
```diff
@@ -167,6 +170,15 @@ class MainActivity : AppCompatActivity(), HasAndroidInjector {
         }
     }
 
+    private fun startCertificatesNestedGraphDestination(navController: NavController, isOnboardingDone: Boolean) {
+        val nestedGraph = navController.graph.findNode(R.id.green_certificate_graph) as NavGraph
+        nestedGraph.startDestination = if (isOnboardingDone) {
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/main/MainActivity.kt:175]: Please use `findNestedGraph(R.id.green_certificate_graph). startDestination `

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/greencertificate/ui/CertificatesSettings.kt [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class CertificatesSettings @Inject constructor(val preferences: CertificatesPreferences) {
+
+    var onboardingStatus: OnboardingStatus
+        get() {
+            val order = preferences.onboardingStatusOrder.value
+            return OnboardingStatus.values().find { it.order == order } ?: OnboardingStatus.NOT_ONBOARDED
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/greencertificate/ui/CertificatesSettings.kt:13]: why do you choose an enum not a boolean?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/greencertificate/ui/onboarding/CertificatesOnboardingFragment.kt [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+            onboardingPrivacy.setOnClickListener {
+                viewModel.onPrivacyButtonPress()
+            }
+        }
+
+        viewModel.routeToScreen.observe2(this) {
+            when (it) {
+
+                CertificatesOnboardingNavigationEvents.NavigateToMainActivity -> {
+                    requireActivity().onBackPressed()
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/greencertificate/ui/onboarding/CertificatesOnboardingFragment.kt:56]: `popBackStack()` ?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/greencertificate/ui/onboarding/CertificatesOnboardingFragment.kt [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+                }
+
+                CertificatesOnboardingNavigationEvents.NavigateToPrivacyFragment -> {
+                    doNavigate(
+                        CertificatesOnboardingFragmentDirections
+                            .actionCertificatesOnboardingFragmentToCheckInPrivacyFragment()
+                    )
+                }
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/greencertificate/ui/onboarding/CertificatesOnboardingFragment.kt:64]: For simplicity can you remove `{}` if it one statement after `-> ` ?

File: Corona-Warn-App/src/main/res/navigation/green_certificate_graph.xml [hunk 5]
```diff
@@ -9,5 +9,33 @@
...
+            android:id="@+id/action_certificatesOnboardingFragment_to_checkInPrivacyFragment"
+            app:destination="@id/checkInPrivacyFragment" />
+        <argument
+            android:name="showBottomNav"
+            android:defaultValue="true"
+            app:argType="boolean" />
+    </fragment>
+
+    <fragment
+        android:id="@+id/checkInPrivacyFragment"
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/res/navigation/green_certificate_graph.xml:36]: why do you need `checkInPrivacyFragment` in this graph ?

File: Corona-Warn-App/src/main/res/layout/fragment_certificates_onboarding.xml [hunk 6]
```diff
@@ -0,0 +1,286 @@
+<?xml version="1.0" encoding="utf-8"?>
+<layout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:bind="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools">
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/res/layout/fragment_certificates_onboarding.xml:5]: Please avoid using this tag 😀 ?

File: Corona-Warn-App/src/main/res/layout/fragment_certificates_onboarding.xml [hunk 7]
```diff
@@ -0,0 +1,286 @@
...
+                    style="@style/subtitleMedium"
+                    android:layout_width="0dp"
+                    android:layout_height="wrap_content"
+                    android:layout_marginTop="31dp"
+                    android:focusable="true"
+                    android:text="@string/green_certificates_onboarding_body"
+                    app:layout_constraintEnd_toEndOf="@id/guideline_end"
+                    app:layout_constraintStart_toStartOf="@id/guideline_start"
+                    app:layout_constraintTop_toBottomOf="@id/onboarding_illustration"
+                    tools:text="Erstellen Sie eine Übersicht über Ihre Kontakte der letzten 14 Tage. So haben Sie bei Bedarf schnell eine vollständige Liste zur Hand." />
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/res/layout/fragment_certificates_onboarding.xml:61]: tools:text is not needed here, you have already 
`android:text="@string/green_certificates_onboarding_body" `

File: Corona-Warn-App/src/main/res/layout/fragment_certificates_onboarding.xml [hunk 8]
```diff
@@ -0,0 +1,286 @@
...
+                    android:id="@+id/onboarding_illustration"
+                    android:layout_width="0dp"
+                    android:layout_height="wrap_content"
+                    android:contentDescription="@string/green_certificates_onboarding_illustration_description"
+                    android:focusable="true"
+                    app:srcCompat="@drawable/ic_green_certificate_onboarding_illustration"
+                    app:layout_constraintEnd_toEndOf="parent"
+                    app:layout_constraintStart_toStartOf="parent"
+                    app:layout_constraintTop_toTopOf="parent"
+                    tools:src="@drawable/ic_contact_diary_illustration_onboarding" />
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/res/layout/fragment_certificates_onboarding.xml:48]: you have  app:srcCompat that provide the image ,and this for  contact diary :)

File: Corona-Warn-App/src/main/res/layout/fragment_certificates_onboarding.xml [hunk 9]
```diff
@@ -0,0 +1,286 @@
...
+
+                <de.rki.coronawarnapp.ui.view.MoreInformationView
+                    android:id="@+id/onboarding_privacy"
+                    android:layout_width="match_parent"
+                    android:layout_height="wrap_content"
+                    android:layout_marginHorizontal="24dp"
+                    android:layout_marginTop="24dp"
+                    android:paddingBottom="10dp"
+                    android:focusable="true"
+                    app:layout_constraintTop_toBottomOf="@+id/check_in_onboarding_card_container"
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/res/layout/fragment_certificates_onboarding.xml:249]: no check-in here

File: Corona-Warn-App/src/main/res/layout/fragment_certificates_onboarding.xml [hunk 10]
```diff
@@ -0,0 +1,286 @@
...
+                    android:layout_width="0dp"
+                    android:layout_height="wrap_content"
+                    android:layout_marginHorizontal="16dp"
+                    android:text="@string/green_certificates_onboarding_stored_in_smartphone"
+                    app:layout_constraintEnd_toEndOf="parent"
+                    app:layout_constraintStart_toEndOf="@id/onboarding_digital_certificates_image"
+                    app:layout_constraintTop_toTopOf="@id/stored_in_smartphone_image" />
+
+                <androidx.constraintlayout.widget.ConstraintLayout
+                    android:id="@+id/check_in_onboarding_card_container"
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/res/layout/fragment_certificates_onboarding.xml:133]: check-in ?

**Review by mtwalli — COMMENTED** [review]
Lgtm 👍🏼  , just small stuff

**Review by jurajkusnier — CHANGES_REQUESTED** [review]
I don't think this is the right solution. Correct me if I'm wrong but from my understanding, we should remove "Add Vaccination Certificate" card and move the functionality to the "Nachweise" tab.
Regarding the onboarding everything is already implemented in the `VaccinationConsentFragment`
`vaccination_consent_fragment`
`VaccinationConsentViewModel`
`VaccinationSettings`
`VaccinationPreferences`
We don't need to reimplement it from scratch, but rather update navigation and new strings.

**Comment by AlexanderAlferov** [comment]
> I don't think this is the right solution. Correct me if I'm wrong but from my understanding, we should remove "Add Vaccination Certificate" card and move the functionality to the "Nachweise" tab.
> Regarding the onboarding everything is already implemented in the `VaccinationConsentFragment`
> `vaccination_consent_fragment`
> `VaccinationConsentViewModel`
> `VaccinationSettings`
> `VaccinationPreferences`
> We don't need to reimplement it from scratch, but rather update navigation and new strings.

Yes, you are right. Missed previous implementation. Will close this PR.

**Pull request closed without merging**